File: presentation/src/main/java/io/tripled/social/client/presentation/console/ConsoleInput.java
package io.tripled.social.client.presentation.console;

import io.tripled.social.client.presentation.Input;

import java.io.BufferedReader;
import java.io.IOException;

public class ConsoleInput implements Input {

  private final BufferedReader bufferedReader;

  public ConsoleInput(BufferedReader bufferedReader) {
    this.bufferedReader = bufferedReader;
  }

  @Override
  public String read() {
    try {
      return bufferedReader.readLine();
    } catch (IOException e) {
      e.printStackTrace();
      return null;
    }
  }
}

File: src/main/kotlin/io/tripled/social/client/presentation/console/ConsoleInput.kt
package io.tripled.social.client.presentation.console

import io.tripled.social.client.presentation.Input
import java.io.BufferedReader
import java.io.IOException

class ConsoleInput(private val bufferedReader: BufferedReader) : Input {
    override fun read(): String? {
        try {
            return bufferedReader.readLine()
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }
    }
}
